adam/learner/integrated_learner.py

```diff
@@ -161,14 +173,18 @@ def observe(
                 # because the static learners do not know how to deal with the temporal
                 # perception graph edge wrappers.
                 # See https://github.com/isi-vista/adam/issues/792 .
-                if not learning_example.perception.is_dynamic():
-                    sub_learner.learn_from(current_learner_state, offset=offset)
-                current_learner_state = sub_learner.enrich_during_learning(
-                    current_learner_state
-                )
-                # Check definiteness after recognizing objects
-                if sub_learner == self.object_learner:
-                    self.learn_definiteness_markers(current_learner_state)
+                try:
+                    if not learning_example.perception.is_dynamic():
+                        sub_learner.learn_from(current_learner_state, offset=offset)
+                    current_learner_state = sub_learner.enrich_during_learning(
+                        current_learner_state
+                    )
+                    # Check definiteness after recognizing objects
+                    if sub_learner == self.object_learner:
+                        self.learn_definiteness_markers(current_learner_state)
+                except Exception as e:  # pylint:disable=broad-except
+                    logging.exception(e)
+                    # pass
```

Comment:
Also, why are we broadly ignoring an exception here?

Reply:
We don't want entire experiments to break because of a single code issue. Exceptions vary a lot - we log it but the comment is necessary for pylint

Reply:
Hm, I'd disagree that we wouldn't want the entire experiment to break because of a single code issue. That code issue could invalidate the results of the experiment because it could hide a bug. 

If there's a specific exception we'd potentially expect to get that could be strictly ignored I'd be more ok to say that ignoring the error is fine (Say we have a specific error type we are using with dotgraph viz) but just broadly ignoring exceptions worries me from Ryan's continual guidance of "Problems should raise runtime errors so they don't hide bugs which are harder to debug!"

Reply:
We still log and debug the exception as needed. This is just a safety belt - we don't have any repeating errors, and we have none for semantics actually, but there were rare errors with e.g one instance in plurals with mislearned object representations. I agree that it's better to keep them as runtime and fix, but it's too rare and infrequent that it's impractical to do otherwise given the timeline.

Reply:
Sure, my concern would be accidentally introducing a new repeating runtime error which is hidden by this exception rather than any current issues. Ideally even mislearned object representations shouldn't cause a runtime error though I would agree it's impractical to fix the rare error you saw given the timeline.